### PR TITLE
Add NovaAPITemplate to be used in NovaSpec

### PR DESCRIPTION
**Pull request by gibizer**
We need to split what input is coming from outside (i.e. from the user) and what input is generated by a controller for our service CRs.

This patch create a NovaAPITemplate for the external input and use that in the NovaSpec instead of the NovaAPISpec.

### CODE REVIEW

File: config/samples/nova_v1beta1_nova-multi-cell.yaml [hunk 1]
```diff
@@ -38,21 +38,17 @@ spec:
...
     defaultConfigOverwrite:
-      policy.yaml: >
-        # my custom policy
-    containerImage: quay.io/tripleowallabycentos9/openstack-nova-api:current-tripleo
+      logging.conf: >
+        # my custom logging configuration
+    # TODO(gibi): How can we have a generic repo + tag definition up in Nova CR
+    # so that we can replace this with auto generated image urls from the Nova
+    # controller?
+    containerImage: quay.io/tripleowallabycentos9/openstack-nova-api:zed-tripleo
```
Comment (SeanMooney) [inline config/samples/nova_v1beta1_nova-multi-cell.yaml:51]: for reasons i dont understand they are publishing each release to a different organisation on quay

quay.io/tripleowallabycentos9 has the wallaby images
https://quay.io/repository/tripleomastercentos9 has the master images.

so the tag might stay current-tripleo to get the latest image but the org would chagne to tripleozedcentos9

can you change this quay.io/tripleowallabycentos9/openstack-nova-api:zed-tripleo to
quay.io/tripleomastercentos9/openstack-nova-api:current-tripleo

**Review by SeanMooney — CHANGES_REQUESTED** [review]
im much happier with this as our interface

**Comment by kk7ds** [comment]
Yeah agreed this is looking cool to me. I think the ContainerImage thing is probably something we should table for the moment and just keep it like you have here. I don't understand how we're going to know if we need to do major or minor upgrade type tasks purely based on updates to an otherwise opaque URL, but presumably that's a larger discussion.

File: config/samples/nova_v1beta1_nova-multi-cell.yaml (hunk 1 above)
Comment (gibizer) [reply config/samples/nova_v1beta1_nova-multi-cell.yaml:51]: Based on the discussion https://coreos.slack.com/archives/CQXJFGMK6/p1663752237261069 I think we will keep an non templated `containerImage` both in each Spec and in each Template too. Version tags cannot be used as offline installation needs to refer images by hash.

File: api/v1beta1/novaapi_types.go [hunk 2]
```diff
@@ -19,12 +19,53 @@ package v1beta1
...
+	// ConfigOverwrite - interface to overwrite default config files like e.g. logging.conf
+	// But can also be used to add additional files. Those get added to the service config dir in /etc/<service> .
+	DefaultConfigOverwrite map[string]string `json:"defaultConfigOverwrite,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// Resources - Compute Resources required by this service (Limits/Requests).
+	// https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+}
+
```
Comment (gibizer) [inline api/v1beta1/novaapi_types.go:68]: We can make KeystoneAuthURL in the Spec required now.

Comment (gibizer) [reply api/v1beta1/novaapi_types.go:68]: Done

File: config/samples/nova_v1beta1_nova-multi-cell.yaml (hunk 1 above)
Comment (SeanMooney) [reply config/samples/nova_v1beta1_nova-multi-cell.yaml:48]: ack i was more pointing out thtat the url you used in the sample was not vaild.

the way tags are used upstream for ooo and downstream fo our product differ so upstream the branch is contoled by the org in the url and the tag is just a a sha or current-tripleo for the head of the branch

so there is no zed-tripleo in tripleowallabycentos9

**Comment by SeanMooney** [comment]
the only nit i have this this is the curren url in the sample is not a real image can you just put it back to the defuatl or use the master url instead.

File: config/samples/nova_v1beta1_nova-multi-cell.yaml (hunk 1 above)
Comment (gibizer) [reply config/samples/nova_v1beta1_nova-multi-cell.yaml]: Ah. OK. Fixed it.

**Review by SeanMooney — APPROVED** [review]

**Pull request merged**